knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -1,51 +1,187 @@
 package consoleclidownload
 
 import (
+	"context"
 	"fmt"
 	"os"
+	"time"
 
 	"github.com/openshift-knative/serverless-operator/knative-operator/pkg/common"
-	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
 
 	mfc "github.com/manifestival/controller-runtime-client"
 	mf "github.com/manifestival/manifestival"
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	consoleroute "github.com/openshift/console-operator/pkg/console/subresource/route"
+	consoleutil "github.com/openshift/console-operator/pkg/console/subresource/util"
+	k8sv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const defaultKnConsoleCLIDownload = "deploy/resources/console_cli_download_kn.yaml"
+const (
+	knDownloadServer                    = "kn-download-server"
+	knConsoleCLIDownloadDeployRoute     = "kn-cli-downloads"
+	knConsoleCLIDownloadDeployNamespace = "kn-cli-downloads"
```

Comment:
As discussed in Slack, I think we agreed on leaving this in knative-serving for now.